### DISCUSSION TITLE
Navbar relative link to external file

### DISCUSSION THREAD

**Discussion by cmoberg** [vuepress/core#261] [Q&A] [answered]
I need to link from the navbar to a file hosted on the same host but not managed by Vuepress.

Something like: `{ text: "API Reference", link: "/api/" }` but where `"/api"` is not mangled.

I understand that links starting with http is treated differently, but I would like to avoid a full URL since the site will be hosted in several different places.

Is there a way to convince vuepress not to mangle a relative link in the navbar?

**Comment by Mister-Hope** [comment, marked as answer]
EDITED: provide full link starting with http would help, and all other links are appended with base. That's the logic.....

It's reasonable for vuepress to treat this link as external, so you should provide the prefix on your own.

**Reply by cmoberg** [reply]
Thanks for quick response @Mister-Hope. Not sure I follow what you're suggesting. I need to use a relative link since the site will be deployed across several domains. E.g. internaldocs.example.com and docs.example.com. This means that if I pick one of them (docs.example.com) the link won't work at internaldocs.example.com.

**Reply by Mister-Hope** [reply]
Detailed explaination:

I am not sure meteorlxy's opinion, but my idea is that it's ok for us to not support it.

The navbar is accpeting absolute link by default, and the default theme treat them as links inside the project. And if you are providing a full link which is not related with this project (http(s):// mailto:// telto://) it will mark it as external and not resolve it with base. You need is much like an edge case outside the navbar, which I think your should write something by your own  to achieve it.

In your need, as you said that it's something outside vuepress project. So what's the problem with setting just with one of your deloy address?

if you real;y want to do so, here are some suggestions:

- make a component (which won't append base for you )using `NavLink` in default theme, and append it in navbar manully
- set a variable called domain, and add it in your config files, build and deploy for each of your domains.

**Reply by Mister-Hope** [reply]
Addtional: some plugins like sitemap and seo which will may ask you set a hostmain and generate files using it. In this case you are epected to set and build in each of your domians.

**Reply by cmoberg** [reply]
Much appreciated answer, thanks! I just wanted to know if there was a way to do it now, and your response is really helpful. I will look into the variable-approach first.
